diff --git a/src/wm-base/linux_lvfs_driver.cpp b/src/wm-base/linux_lvfs_driver.cpp
index 882b851..a38b470 100644
--- a/src/wm-base/linux_lvfs_driver.cpp
+++ b/src/wm-base/linux_lvfs_driver.cpp
@@ -63,7 +63,7 @@ namespace LocalDriver
 	{
 		if( !isRunning() )
 		{
-			_rootDirectory = generalSettings.appStoragePath;
+			_rootDirectory = generalSettings.m_appStoragePath;
 			_mountPoint = QDir::homePath() + QDir::separator() + QString::fromUtf8("WebMounter") + QDir::separator();
 			QDir dir;
 			dir.mkpath(_mountPoint);
diff --git a/src/wm-plugins/wm-plugins.pro b/src/wm-plugins/wm-plugins.pro
index b812eb9..6936a97 100644
--- a/src/wm-plugins/wm-plugins.pro
+++ b/src/wm-plugins/wm-plugins.pro
@@ -1,3 +1,3 @@
 TEMPLATE = subdirs
-SUBDIRS += facebook vk yaf ya-narod ya-disk
+SUBDIRS += vk yaf ya-narod ya-disk
 
