diff -ru xmahjong/draw.c xmahjong.new/draw.c
--- xmahjong/draw.c	1990-09-14 15:02:30.000000000 +0200
+++ xmahjong.new/draw.c	2006-08-10 12:04:06.000000000 +0200
@@ -8,6 +8,8 @@
  */
 
 #include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
 #include <X11/Xlib.h>
 #include <X11/Xutil.h>
 #include <sys/time.h>
diff -ru xmahjong/initial.c xmahjong.new/initial.c
--- xmahjong/initial.c	1995-06-10 13:25:58.000000000 +0200
+++ xmahjong.new/initial.c	2006-08-10 11:56:31.000000000 +0200
@@ -10,6 +10,7 @@
 #include <netdb.h>
 #include <stdio.h>
 #include <signal.h>
+#include <string.h>
 #include <sys/time.h>
 #include <sys/types.h>
 #include <sys/socket.h>
diff -ru xmahjong/packet.c xmahjong.new/packet.c
--- xmahjong/packet.c	1990-09-14 15:02:32.000000000 +0200
+++ xmahjong.new/packet.c	2006-08-10 12:05:53.000000000 +0200
@@ -9,6 +9,8 @@
 
 #include <stdio.h>
 #include <signal.h>
+#include <string.h>
+#include <unistd.h>
 #include <sys/time.h>
 #include <sys/types.h>
 #include <sys/socket.h>
diff -ru xmahjong/xmahjongg.c xmahjong.new/xmahjongg.c
--- xmahjong/xmahjongg.c	1995-06-21 17:34:00.000000000 +0200
+++ xmahjong.new/xmahjongg.c	2006-08-10 12:02:48.000000000 +0200
@@ -9,6 +9,8 @@
 
 #include <stdio.h>
 #include <string.h>
+#include <stdlib.h>
+#include <unistd.h>
 #include <X11/Xlib.h>
 #include <X11/Xutil.h>
 #include "xmahjongg.h"
