--- monsterz.py 2007/12/17 22:04:29 135
+++ monsterz.py 2008/05/09 16:31:26 137
@@ -108,7 +108,9 @@
                 M = int(max(r, g, b))
                 m = int(min(r, g, b))
                 val = (2 * M + r + g + b) / 5
-                p[:] = (val + r) / 2, (val + g) / 2, (val + b) / 2
+                p[0] = (val + r) / 2
+                p[1] = (val + g) / 2
+                p[2] = (val + b) / 2
                 if alpha[y][x] >= 250:
                     alpha[y][x] = 255 - (M - m) * 3 / 4
         del pixels
@@ -129,7 +131,9 @@
                 r, g, b = p
                 M = int(max(r, g, b))
                 m = int(min(r, g, b))
-                p[:] = (m + r) / 2, (m + g) / 2, (m + b) / 2
+                p[0] = (m + r) / 2
+                p[1] = (m + g) / 2
+                p[2] = (m + b) / 2
                 if alpha[y][x] >= 250:
                     alpha[y][x] = 255 - M * 2 / 3
         del pixels
