diff --git a/src/plugins/snails/concurrentexceptions.h b/src/plugins/snails/concurrentexceptions.h
index 9ece730..c776949 100644
--- a/src/plugins/snails/concurrentexceptions.h
+++ b/src/plugins/snails/concurrentexceptions.h
@@ -57,7 +57,7 @@ namespace Snails
 		const QByteArray What_;
 	public:
 		InvokeFailedException (const TaskQueueItem&);
-		~InvokeFailedException () noexcept = default;
+		~InvokeFailedException () noexcept {};
 
 		const char* what () const noexcept override;
 	};
@@ -67,7 +67,7 @@ namespace Snails
 		const QString Message_;
 	public:
 		AuthorizationException (const QString&);
-		~AuthorizationException () noexcept = default;
+		~AuthorizationException () noexcept {};
 
 		const QString& GetMessage () const;
 	};
@@ -76,7 +76,7 @@ namespace Snails
 	{
 	public:
 		TimeoutException () = default;
-		~TimeoutException () noexcept = default;
+		~TimeoutException () noexcept {};
 	};
 
 	template<typename Wrapped>
@@ -89,7 +89,7 @@ namespace Snails
 		{
 		}
 
-		~WrappedException () noexcept = default;
+		~WrappedException () noexcept {};
 
 		const char* what () const noexcept override
 		{
