--- src/plugins/bittorrent/core.h
+++ src/plugins/bittorrent/core.h
@@ -29,6 +29,7 @@
 
 #pragma once
 
+#ifndef Q_MOC_RUN
 #include <map>
 #include <list>
 #include <deque>
@@ -42,12 +43,13 @@
 #include <libtorrent/torrent_handle.hpp>
 #include <libtorrent/session_status.hpp>
 #include <libtorrent/session.hpp>
-#include <interfaces/iinfo.h>
-#include <interfaces/structures.h>
 #include <util/tags/tagscompletionmodel.h>
 #include "torrentinfo.h"
 #include "fileinfo.h"
 #include "peerinfo.h"
+#endif
+#include <interfaces/iinfo.h>
+#include <interfaces/structures.h>
 
 class QTimer;
 class QDomElement;
--- src/plugins/bittorrent/torrentfilesmodel.cpp
+++ src/plugins/bittorrent/torrentfilesmodel.cpp
@@ -29,7 +29,9 @@
 
 #include "torrentfilesmodel.h"
 #include <iterator>
+#ifndef Q_MOC_RUN
 #include <boost/functional/hash.hpp>
+#endif
 #include <QUrl>
 #include <QTimer>
 #include <QtDebug>
--- src/plugins/netstoremanager/fileswatcher_inotify.h
+++ src/plugins/netstoremanager/fileswatcher_inotify.h
@@ -29,7 +29,9 @@
 
 #pragma once
 
+#ifndef Q_MOC_RUN
 #include <boost/bimap.hpp>
+#endif
 #include <QObject>
 #include <QStringList>
 #include <QMultiMap>
