diff --git a/src/plugins/poshuku/plugins/dcac/effects.h b/src/plugins/poshuku/plugins/dcac/effects.h
index 41cbd5a..5b67a6f 100644
--- a/src/plugins/poshuku/plugins/dcac/effects.h
+++ b/src/plugins/poshuku/plugins/dcac/effects.h
@@ -30,7 +30,9 @@
 #pragma once
 
 #include <QMetaType>
+#ifndef Q_MOC_RUN
 #include <boost/variant.hpp>
+#endif
 
 namespace LeechCraft
 {
diff --git a/src/plugins/poshuku/plugins/dcac/tests/testbase.h b/src/plugins/poshuku/plugins/dcac/tests/testbase.h
index 49e86dc..6cf34f9 100644
--- a/src/plugins/poshuku/plugins/dcac/tests/testbase.h
+++ b/src/plugins/poshuku/plugins/dcac/tests/testbase.h
@@ -36,7 +36,9 @@
 #include <QMap>
 #include <QElapsedTimer>
 #include <QtDebug>
+#ifndef Q_MOC_RUN
 #include <util/sll/qtutil.h>
+#endif
 
 inline bool operator< (const QSize& s1, const QSize& s2)
 {
