--- Makefile.orig	2014-01-29 03:06:48.000000000 +0300
+++ Makefile	2014-02-03 22:17:06.547122520 +0300
@@ -66,8 +66,8 @@
 make:
 	@printf "\033[1;31mCompile edid-decode as inex-decode \033[0m$1\n"
 	$(CC) -o i-nex-edid JSON/edid-decode.c $(CFLAGS) $(additional_confflags)
-	$(CC) -o i-nex-cpuid JSON/cpu.c -static $(CC_OPTS_LIBCPUID) $(additional_confflags)
-	$(CC) -o i-nex-usage JSON/free.c -static $(CC_OPTS_LIBPROCPS) $(additional_confflags)
+	$(CC) -o i-nex-cpuid JSON/cpu.c $(CC_OPTS_LIBCPUID) $(additional_confflags)
+	$(CC) -o i-nex-usage JSON/free.c $(CC_OPTS_LIBPROCPS) $(additional_confflags)
 	$(CC) -o i-nex-cpusage JSON/cpusage.c $(additional_confflags)
 	$(CC) -o i-nex-glinfo JSON/glinfo.c $(libx11) $(libgl) $(libxext) $(additional_confflags)
 	$(CC) -o i-nex-uname JSON/uname.c $(additional_confflags)
