diff -U 3 -H -d -r -N -- i-nex_orig/src/i-nex/.src/FReport_Gen.class i-nex/src/i-nex/.src/FReport_Gen.class
--- i-nex_orig/src/i-nex/.src/FReport_Gen.class	2013-01-01 23:27:04.000000000 +0100
+++ i-nex/src/i-nex/.src/FReport_Gen.class	2013-02-02 23:49:41.281846446 +0100
@@ -16,6 +16,7 @@
   ComboBox1.Add("paste.drizzle.org")
   ComboBox1.Add("paste.kde.org")
   ComboBox1.Add("paste.openstack.org")
+  ComboBox1.Add("susepaste.org")
   ComboBox1.Add("paste.pocoo.org")
   ComboBox1.Add("paste.pound-python.org")
   ComboBox1.Add("paste.ubuntu.com")
diff -U 3 -H -d -r -N -- i-nex_orig/src/i-nex/.src/Finfosys.class i-nex/src/i-nex/.src/Finfosys.class
