--- src/core.cpp.orig
+++ src/core.cpp
@@ -1698,6 +1698,8 @@
 	if (pref->ao != "player_default") {
 		if (!pref->ao.isEmpty()) {
 			proc->setOption("ao", pref->ao );
+		} else {
+			proc->setOption("ao", "@@DEFAULT@@," );
 		}
 	}
 
--- src/prefgeneral.cpp.orig
+++ src/prefgeneral.cpp
@@ -183,6 +183,9 @@
 
 	QString ao = pref->ao;
 
+	if (ao.isEmpty())
+		ao = "@@DEFAULT@@,";
+
 #ifdef Q_OS_OS2
 	if (ao.isEmpty()) {
 		if (pref->mplayer_detected_version >= MPLAYER_KAI_VERSION) {
