--- bear-engine/core/src/universe/code/zone.cpp.orig	2011-07-17 22:01:25.000000000 +0300
+++ bear-engine/core/src/universe/code/zone.cpp	2013-08-04 00:03:23.000000000 +0300
@@ -1,33 +1,20 @@
 /*
-  Bear Engine
+  Copyright (C) 2012 Stuffomatic Ltd. <contact@stuff-o-matic.com>
 
-  Copyright (C) 20052011 Julien Jorge, Sebastien Angibaud
+  All rights reserved.
 
-  This program is free software; you can redistribute it and/or modify it
-  under the terms of the GNU General Public License as published by the
-  Free Software Foundation; either version 2 of the License, or (at your
-  option) any later version.
-
-  This program is distributed in the hope that it will be useful, but WITHOUT
-  ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
-  FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
-  more details.
-
-  You should have received a copy of the GNU General Public License along
-  with this program; if not, write to the Free Software Foundation, Inc.,
-  51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
-
-  contact: plee-the-bear@gamned.org
-
-  Please add the tag [Bear] in the subject of your mails.
+  See the accompanying license file for details about usage, modification and
+  distribution of this file.
 */
 /**
- * \file zone.cpp
+ * \file
  * \brief Implementation of the bear::universe::zone class.
  * \author Julien Jorge
  */
 #include "universe/zone.hpp"
 
+#include <claw/exception.hpp>
+
 /*----------------------------------------------------------------------------*/
 const unsigned int bear::universe::zone::cardinality = 9;
 
@@ -96,4 +83,8 @@
   case bottom_zone: return top_zone;
   case bottom_right_zone: return top_left_zone;
   }
+
+  // this should never happen but it turns a warning off
+  throw new claw::exception
+    ( "bear::universe::zone::opposite_of(): Invalid position size" );
 } // zone::opposite_of()
