Index: nonints.h
===================================================================
RCS file: /volume/cvs/bmake/nonints.h,v
retrieving revision 1.15
diff -u -p -r1.15 nonints.h
--- nonints.h	27 Sep 2016 20:40:54 -0000	1.15
+++ nonints.h	18 Jul 2017 23:43:15 -0000
@@ -143,6 +143,11 @@ int Str_Match(const char *, const char *
 char *Str_SYSVMatch(const char *, const char *, int *len);
 void Str_SYSVSubst(Buffer *, char *, char *, int);
 
+#ifndef HAVE_STRLCPY
+/* strlcpy.c */
+size_t strlcpy(char *, const char *, size_t);
+#endif
+
 /* suff.c */
 void Suff_ClearSuffixes(void);
 Boolean Suff_IsTransform(char *);
