diff --git a/data/xfdashboard-settings.desktop.in.in b/data/xfdashboard-settings.desktop.in.in
index 1011c0e..1f329a4 100644
--- a/data/xfdashboard-settings.desktop.in.in
+++ b/data/xfdashboard-settings.desktop.in.in
@@ -3,8 +3,8 @@ Version=1.0
 Type=Application
 _Name=Xfdashboard settings
 _Comment=Configure xfdashboard
-TryExec=@bindir@/xfdashboard-settings
-Exec=@bindir@/xfdashboard-settings
+TryExec=xfdashboard-settings
+Exec=xfdashboard-settings
 Icon=xfdashboard
 Categories=XFCE;GTK;Settings;DesktopSettings;X-XFCE-SettingsDialog;X-XFCE-PersonalSettings;
 OnlyShowIn=XFCE;
diff --git a/data/xfdashboard.desktop.in.in b/data/xfdashboard.desktop.in.in
index fd63ac0..d7bd368 100644
--- a/data/xfdashboard.desktop.in.in
+++ b/data/xfdashboard.desktop.in.in
@@ -2,8 +2,8 @@
 Type=Application
 _Name=Xfdashboard
 _Comment=Maybe a Gnome shell like dashboard for Xfce
-TryExec=@bindir@/xfdashboard
-Exec=@bindir@/xfdashboard
+TryExec=xfdashboard
+Exec=xfdashboard
 Icon=xfdashboard
 Categories=System;Applet;
 NotShowIn=GNOME;KDE;Unity;
