--- Tools/XMLConverter/CMakeLists.txt.orig	2012-09-02 07:27:16.000000000 +0200
+++ Tools/XMLConverter/CMakeLists.txt	2012-09-23 01:36:49.882143240 +0200
@@ -13,22 +13,16 @@ set(HEADER_FILES
   include/OgreXMLMeshSerializer.h
   include/OgreXMLPrerequisites.h
   include/OgreXMLSkeletonSerializer.h
-  include/tinystr.h
-  include/tinyxml.h
 )
 
 set(SOURCE_FILES 
   src/main.cpp
   src/OgreXMLMeshSerializer.cpp
   src/OgreXMLSkeletonSerializer.cpp
-  src/tinystr.cpp
-  src/tinyxml.cpp
-  src/tinyxmlerror.cpp
-  src/tinyxmlparser.cpp
 )
 
 include_directories(${CMAKE_CURRENT_SOURCE_DIR}/include)
 add_definitions(-DTIXML_USE_STL)
 ogre_add_executable(OgreXMLConverter ${HEADER_FILES} ${SOURCE_FILES})
-target_link_libraries(OgreXMLConverter ${OGRE_LIBRARIES})
+target_link_libraries(OgreXMLConverter ${OGRE_LIBRARIES} -ltinyxml)
 ogre_config_tool(OgreXMLConverter)
