--- CMakeLists.txt.orig	2015-07-04 05:50:52.000000000 +0300
+++ CMakeLists.txt	2016-07-06 20:29:07.270508000 +0300
@@ -7,8 +7,8 @@
 OPTION(DEMO "Build demo version" off)
 
 if(KCM_AUDIO)
-	set(CMAKE_C_FLAGS "-DKCM_AUDIO")
-	set(CMAKE_CXX_FLAGS "-DKCM_AUDIO")
+	set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -DKCM_AUDIO")
+	set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DKCM_AUDIO")
 endif(KCM_AUDIO)
 
 # function to copy files to build dir:
