--- CMakeLists.txt.orig	2015-07-04 05:50:52.000000000 +0300
+++ CMakeLists.txt	2016-07-06 20:29:07.270508000 +0300
@@ -8,8 +8,8 @@
 OPTION(STEAMWORKS "Enable Steamworks" off)
 
 if(KCM_AUDIO)
-	set(CMAKE_C_FLAGS "-DKCM_AUDIO")
-	set(CMAKE_CXX_FLAGS "-DKCM_AUDIO")
+	set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -DKCM_AUDIO")
+	set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DKCM_AUDIO")
 endif()
 
 # function to copy files to build dir:
--- CMakeLists.txt.orig	2016-06-04 00:06:00.000000000 +0300
+++ CMakeLists.txt	2016-07-08 06:44:05.417884840 +0300
@@ -246,7 +246,17 @@
 	set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -Wl,-rpath,\$ORIGIN")
 	SET(MAIN_LIBS 
 		mo2-engine
-		allegro_monolith${ALLEG_SUFFIX}
+		allegro_ttf${ALLEG_SUFFIX}
+		allegro_font${ALLEG_SUFFIX}
+		allegro_primitives${ALLEG_SUFFIX}
+		allegro_dialog${ALLEG_SUFFIX}
+		allegro_memfile${ALLEG_SUFFIX}
+		allegro_image${ALLEG_SUFFIX}
+		allegro_physfs${ALLEG_SUFFIX}
+		allegro${ALLEG_SUFFIX}
+		allegro_main${ALLEG_SUFFIX}
+		allegro_acodec${ALLEG_SUFFIX}
+		allegro_audio${ALLEG_SUFFIX}
 		${EXTRA_LIBS}
 		Xcursor Xi Xinerama Xrandr SM ICE X11 Xext
 	)
--- CMakeLists.txt.orig	2015-01-09 19:57:30.000000000 +0300
+++ CMakeLists.txt	2015-05-19 23:17:40.253461120 +0300
@@ -259,10 +259,8 @@
 	if(RASPBERRYPI)
 		LIST(APPEND MAIN_LIBS GLESv2 EGL bcm_host m pthread /usr/local/lib/libfreetype.a physfs /usr/local/lib/libpng.a /usr/local/lib/libz.a lua5.2)
 	else()
-		# trentg's build
-		LIST(APPEND MAIN_LIBS GL GLU /usr/local/lib/libphysfs.a /usr/local/lib/libfreetype.a /usr/local/lib/libz.a /usr/local/lib/libpng.a /usr/local/lib/liblua5.2.a pthread)
 		# normal build
-		#LIST(APPEND MAIN_LIBS GL GLU physfs freetype z png lua5.2)
+		LIST(APPEND MAIN_LIBS GL GLU physfs freetype z png lua5.2)
 	endif()
 endif()
 endif()
