--- CMakeLists.txt.orig	2015-01-09 19:57:30.000000000 +0300
+++ CMakeLists.txt	2015-05-19 23:17:40.253461120 +0300
@@ -270,10 +270,8 @@
 	if(RASPBERRYPI)
 		LIST(APPEND MAIN_LIBS GLESv2 EGL bcm_host m pthread /usr/local/lib/libfreetype.a physfs /usr/local/lib/libpng.a /usr/local/lib/libz.a lua5.2)
 	else(RASPBERYPI)
-		# trentg's build
-		LIST(APPEND MAIN_LIBS GL GLU /usr/local/lib/libphysfs.a /usr/local/lib/libfreetype.a /usr/local/lib/libz.a /usr/local/lib/libpng.a /usr/local/lib/liblua5.2.a)
 		# normal build
-		#LIST(APPEND MAIN_LIBS GL GLU physfs freetype z png lua5.2)
+		LIST(APPEND MAIN_LIBS GL GLU physfs freetype z png lua5.2)
 	endif(RASPBERRYPI)
 endif(APPLE)
 endif(WIN32)
