--- Makefile.orig	2012-03-22 22:28:34.000000000 +0300
+++ Makefile	2012-04-21 23:55:14.027000076 +0300
@@ -1,6 +1,6 @@
 GAMENAME = WarlocksGauntlet
 
-BINARY = $(GAMENAME).bin32
+BINARY = warlocksgauntlet
 
 -include Makefile.sources
 
@@ -9,9 +9,9 @@
     CPPFLAGS += -DSCM_REVISION="$(SCM_REVISION)"
 endif
 
-LDFLAGS = -L"./libs32"
+# LDFLAGS = -L"./libs32"
 
-LIBS = -lsfml-graphics -lsfml-window -lsfml-system -lsfml-audio
+LIBS = -lsfml-graphics -lsfml-window -lsfml-system -lsfml-audio -lopenal -lGL -lGLEW -lz
 
 #default:
 CXXFLAGS = -g -Wall
