--- src/Map/CMapManager.cpp.orig	2012-03-08 19:37:30.000000000 +0300
+++ src/Map/CMapManager.cpp	2012-03-28 19:03:19.000000000 +0300
@@ -47,8 +47,13 @@
         mWorldGraph = new CWorldGraph();
         mWorldGraph->LoadFromFile("data/maps/world-graph.xml");
 
-        if (!boost::filesystem::exists(mWorld))
-            boost::filesystem::create_directories(mWorld);
+        if (!boost::filesystem::exists(mWorld)) {
+            try {
+                boost::filesystem::create_directories(mWorld);
+            } catch (const std::exception & e) {
+                fprintf(stderr, "ERROR: failed to create %s directories, what: %s\n", mWorld.c_str(), e.what());
+            }
+        }
 	}
 
 	CMapManager::~CMapManager()
