--- src/Map/CMapImageGenerator.h.orig	2012-03-22 22:28:46.000000000 +0300
+++ src/Map/CMapImageGenerator.h	2012-03-25 22:50:44.130677774 +0300
@@ -2,6 +2,9 @@
 #include <SFML/Graphics.hpp>
 #include "CWorldGraph.h"
 #include "../Utils/CSingleton.h"
+#include <map>
+#include <vector>
+#include <string>
 
 #define gMapImageGenerator Map::CMapImageGenerator::GetSingleton()
 
@@ -56,8 +59,8 @@
     std::string startingMap;
     std::string startingRegion;
     std::map<std::string, CWorldGraphMapEx> maps;
-	std::map<std::string, std::vector<std::string>> schemes;
-	std::map<std::string, std::vector<std::string>> schemeConvexHulls;
+	std::map<std::string, std::vector<std::string> > schemes;
+	std::map<std::string, std::vector<std::string> > schemeConvexHulls;
 	CWorldGraphEx() { };
     void BuildFromWorldGraph();
 };
