--- Makefile.x86_64.orig	2012-03-22 22:28:34.000000000 +0300
+++ Makefile.x86_64	2012-04-21 23:55:11.225000076 +0300
@@ -2,7 +2,7 @@
 
 ARCH = "x86_64"
 
-BINARY = $(GAMENAME).bin64
+BINARY = warlocksgauntlet
 
 -include Makefile.sources
 
@@ -11,9 +11,9 @@
     CPPFLAGS += -DSCM_REVISION="$(SCM_REVISION)"
 endif
 
-LDFLAGS = -L"./libs64"
+# LDFLAGS = -L"./libs64"
 
-LIBS = -lsfml-graphics -lsfml-window -lsfml-system -lsfml-audio -lopenal
+LIBS = -lsfml-graphics -lsfml-window -lsfml-system -lsfml-audio -lopenal -lGL -lGLEW -lz
 
 #default:
 CXXFLAGS = -g -Wall -m64
