--- Makefile.x86_64.orig	2012-03-22 22:28:34.000000000 +0300
+++ Makefile.x86_64	2012-03-26 19:29:19.725000034 +0300
@@ -2,7 +2,7 @@
 
 ARCH = "x86_64"
 
-BINARY = $(GAMENAME).bin64
+BINARY = warlocksgauntlet
 
 -include Makefile.sources
 
@@ -13,7 +13,7 @@
 
 LDFLAGS = -L"./libs64"
 
-LIBS = -lsfml-graphics -lsfml-window -lsfml-system -lsfml-audio -lopenal
+LIBS = -lsfml-graphics -lsfml-window -lsfml-system -lsfml-audio -lopenal -lGL
 
 #default:
 CXXFLAGS = -g -Wall -m64
