--- src/Makefile.am.orig	2013-10-24 15:11:14.000000000 +0300
+++ src/Makefile.am	2013-10-24 23:35:46.000000000 +0300
@@ -49,10 +49,10 @@
 
 # This rule lets GNU make create any moc_*.cpp from the equivalent *.h
 moc_%.cpp: %.h
-	moc-qt4 $< -o $@
+	`pkg-config --variable=exec_prefix QtCore`/bin/moc $< -o $@
 
 ui_%.h: %.ui
-	uic-qt4 $< -o $@
+	`pkg-config --variable=exec_prefix QtCore`/bin/uic $< -o $@
 
 # Adjust this line according to your resources.qrc
 resources.cpp : resources.qrc octave-gui.png
--- src/Makefile.in.orig	2013-10-24 15:11:19.000000000 +0300
+++ src/Makefile.in	2013-10-24 23:35:45.000000000 +0300
@@ -1027,10 +1027,10 @@
 
 # This rule lets GNU make create any moc_*.cpp from the equivalent *.h
 moc_%.cpp: %.h
-	moc-qt4 $< -o $@
+	`pkg-config --variable=exec_prefix QtCore`/bin/moc $< -o $@
 
 ui_%.h: %.ui
-	uic-qt4 $< -o $@
+	`pkg-config --variable=exec_prefix QtCore`/bin/uic $< -o $@
 
 # Adjust this line according to your resources.qrc
 resources.cpp : resources.qrc octave-gui.png
--- src/main.cpp.orig	2013-10-24 23:53:54.000000000 +0300
+++ src/main.cpp	2013-10-24 23:53:54.000000000 +0300
@@ -19,7 +19,7 @@
 #include <unistd.h>
 
 static const KLocalizedString description =ki18n("kXneur (KDE X Neural Switcher) is xNeur front-end for KDE ( http://xneur.ru ).\nThis version work with XNeur v.0.16 only");
-static const char version[] = "0.16.0";
+static const char version[] = "0.17.0";
 
 int main(int argc, char *argv[])
 {
