diff --git a/src/plugins/azoth/actionsmanager.cpp b/src/plugins/azoth/actionsmanager.cpp
index 8c11a81..f0efc85 100644
--- a/src/plugins/azoth/actionsmanager.cpp
+++ b/src/plugins/azoth/actionsmanager.cpp
@@ -424,7 +424,7 @@ namespace Azoth
 
 			new Util::DelayedExecutor
 			{
-				[w, imjw, accObj] () -> void
+				[=] () -> void
 				{
 					imjw->Join (accObj);
 					w->deleteLater ();
diff --git a/src/plugins/azoth/plugins/xoox/clientconnection.cpp b/src/plugins/azoth/plugins/xoox/clientconnection.cpp
index cae3b05..e0d7cce 100644
--- a/src/plugins/azoth/plugins/xoox/clientconnection.cpp
+++ b/src/plugins/azoth/plugins/xoox/clientconnection.cpp
@@ -998,7 +998,7 @@ namespace Xoox
 			{
 				new Util::DelayedExecutor
 				{
-					[] () -> void
+					[this] () -> void
 					{
 						const auto& e = Util::MakeNotification ("Azoth",
 								tr ("Using the juick.com service is discouraged. Please consider "
