--- KKEdit/resources/pixmaps/KKEditRoot.desktop.orig	2015-05-18 20:44:33.679338200 +0300
+++ KKEdit/resources/pixmaps/KKEditRoot.desktop	2015-05-18 21:15:38.667338200 +0300
@@ -2,9 +2,10 @@
 Version=1.0
 Encoding=UTF-8
 Name=KKEdit - Root
+GenericName=Advanced Text Editor - Root Mode
 Comment=A GTK2 text editor inspired ( but not based on ) BBEdit.
 Terminal=false
 Type=Application
-Categories=System;
-MimeType=text/plain
-Icon=KKEditRoot.png
+Categories=Utility;TextEditor;GTK;
+MimeType=text/plain;
+Icon=KKEditRoot
--- KKEdit/resources/pixmaps/KKEdit.desktop.orig	2015-05-18 20:44:34.720338200 +0300
+++ KKEdit/resources/pixmaps/KKEdit.desktop	2015-05-18 21:15:37.611338200 +0300
@@ -2,10 +2,11 @@
 Version=1.0
 Encoding=UTF-8
 Name=KKEdit
+GenericName=Advanced Text Editor
 Comment=A GTK2 text editor inspired ( but not based on ) BBEdit.
 Terminal=false
 Type=Application
-Categories=Utility;
-MimeType=text/plain
-Icon=KKEdit.png
+Categories=Utility;TextEditor;GTK;
+MimeType=text/plain;
+Icon=KKEdit
 Exec=kkedit %f
