Index: src/CMakeLists.txt
===================================================================
--- src/CMakeLists.txt.orig
+++ src/CMakeLists.txt
@@ -11,6 +11,24 @@
 macro_optional_find_package(Nepomuk)
 macro_optional_find_package(QJSON)
 
+execute_process(
+  COMMAND pkg-config --variable=mltdatadir mlt-framework
+  OUTPUT_VARIABLE MLT_DATADIR
+  RESULT_VARIABLE MLT_DATADIR_failed)
+if (NOT MLT_DATADIR_failed)
+  string(REGEX REPLACE "[\r\n]" "" MLT_DATADIR "${MLT_DATADIR}")
+  add_definitions(-DMLT_DATADIR=\\\"\"${MLT_DATADIR}\"\\\")
+endif(NOT MLT_DATADIR_failed)
+
+execute_process(
+  COMMAND pkg-config --variable=meltbin mlt-framework
+  OUTPUT_VARIABLE MLT_MELTBIN
+  RESULT_VARIABLE MLT_MELTBIN_failed)
+if (NOT MLT_MELTBIN_failed)
+  string(REGEX REPLACE "[\r\n]" "" MLT_MELTBIN "${MLT_MELTBIN}")
+  add_definitions(-DMLT_MELTBIN=\\\"\"${MLT_MELTBIN}\"\\\")
+endif(NOT MLT_MELTBIN_failed)
+
 option(WITH_JogShuttle "Build Jog/Shuttle support" ON)
 
 set(FFMPEG_SUFFIX "" CACHE STRING "FFmpeg custom suffix")

Index: src/mainwindow.cpp
===================================================================
--- src/mainwindow.cpp.orig
+++ src/mainwindow.cpp
@@ -2359,11 +2359,11 @@
     }
 
     if (KdenliveSettings::mltpath().isEmpty()) {
-        KdenliveSettings::setMltpath(QString(MLT_PREFIX) + QString("/share/mlt/profiles/"));
+        KdenliveSettings::setMltpath(QString(MLT_DATADIR) + QString("/profiles/"));
     }
 
     if (KdenliveSettings::rendererpath().isEmpty() || KdenliveSettings::rendererpath().endsWith("inigo")) {
-        QString meltPath = QString(MLT_PREFIX) + QString("/bin/melt");
+        QString meltPath = QString(MLT_MELTBIN);
         if (!QFile::exists(meltPath)) {
             meltPath = KStandardDirs::findExe("melt");
         }
