diff -crB modem-manager-gui-0.0.16/src/notifications.c modem-manager-gui-0.0.16.new/src/notifications.c
*** modem-manager-gui-0.0.16/src/notifications.c	2013-06-23 22:07:00.000000000 +0400
--- modem-manager-gui-0.0.16.new/src/notifications.c	2013-07-28 16:54:42.000000000 +0400
***************
*** 30,35 ****
--- 30,36 ----
  mmgui_notifications_t mmgui_notifications_new(mmgui_libpaths_cache_t libcache)
  {
  	mmgui_notifications_t notifications;
+ 	GError *error;
  	gboolean libopened;
  	GtkSettings *gtksettings;
  	gchar *gtksoundtheme;
***************
*** 68,73 ****
--- 69,81 ----
  		} else {
  			//Initialize libnotify
  			(notifications->notify_init)("Modem Manager GUI");
+ 			/*Load icon for notifications*/
+ 			error = NULL;
+ 			notifications->notifyicon = gdk_pixbuf_new_from_file(RESOURCE_MAINWINDOW_ICON, &error);
+ 			if ((notifications->notifyicon == NULL) && (error != NULL)) {
+ 				g_debug("Error loadig application icon: %s\n", error->message);
+ 				g_error_free(error);
+ 			}
  		}
  	}
  	
***************
*** 130,135 ****
--- 138,144 ----
  		notification = (notifications->notify_notification_new)(caption, text, NULL);
  		if (notification != NULL) {
  			(notifications->notify_notification_set_timeout)(notification, 3000);
+ 			(notifications->notify_notification_set_image_from_pixbuf)(notification, notifications->notifyicon);
  			(notifications->notify_notification_show)(notification, NULL);
  		}
  	}
***************
*** 171,176 ****
--- 180,189 ----
  		//Then unload module
  		g_module_close(notifications->notifymodule);
  		notifications->notifymodule = NULL;
+ 		/*And free icon*/
+ 		if (notifications->notifyicon != NULL) {
+ 			g_object_unref(notifications->notifyicon);
+ 		}
  	}
  	
  	if (notifications->canberramodule != NULL) {
diff -crB modem-manager-gui-0.0.16/src/notifications.h modem-manager-gui-0.0.16.new/src/notifications.h
*** modem-manager-gui-0.0.16/src/notifications.h	2013-06-23 21:49:52.000000000 +0400
--- modem-manager-gui-0.0.16.new/src/notifications.h	2013-07-28 16:50:08.000000000 +0400
***************
*** 80,85 ****
--- 80,87 ----
  	notify_notification_set_category_func notify_notification_set_category;
  	notify_notification_set_urgency_func notify_notification_set_urgency;
  	notify_notification_show_func notify_notification_show;
+ 	/*notifications icon*/
+ 	GdkPixbuf *notifyicon;
  	//libcanberra functions
  	ca_context_create_func ca_context_create;
  	ca_context_destroy_func ca_context_destroy;
