diff -crB modem-manager-gui-0.0.16/configure modem-manager-gui-0.0.16.new/configure
*** modem-manager-gui-0.0.16/configure	2013-07-16 13:47:50.000000000 +0400
--- modem-manager-gui-0.0.16.new/configure	2013-07-28 01:12:19.847882437 +0400
***************
*** 156,162 ****
  Comment[pt_BR]=Ferramenta para controle de funções específicas de modem EDGE/3G/4G
  Comment[tr]=EDGE/3G/4G modemler için özel işlev kontrol aracı
  Comment[id]=Perangkat lunak untuk mengendalikan fungsi spesifik modem EDGE/3G/4G
! Exec=$prefix/bin/modem-manager-gui
  Icon=$prefix/share/pixmaps/modem-manager-gui.png
  Terminal=false
  Type=Application
--- 156,162 ----
  Comment[pt_BR]=Ferramenta para controle de funções específicas de modem EDGE/3G/4G
  Comment[tr]=EDGE/3G/4G modemler için özel işlev kontrol aracı
  Comment[id]=Perangkat lunak untuk mengendalikan fungsi spesifik modem EDGE/3G/4G
! Exec=$prefix/bin/modem-manager-gui --mmmodule=mm06
  Icon=$prefix/share/pixmaps/modem-manager-gui.png
  Terminal=false
  Type=Application
