diff -up plee-the-bear-0.6.0/CMakeLists.txt.svnclawfix plee-the-bear-0.6.0/CMakeLists.txt
--- plee-the-bear-0.6.0/CMakeLists.txt.svnclawfix	2011-08-25 16:26:17.860303496 -0400
+++ plee-the-bear-0.6.0/CMakeLists.txt	2011-08-25 16:26:59.997753959 -0400
@@ -81,8 +81,6 @@ macro(FIND_SVN_CLAW)
   # find the module
   find_package(libclaw)
 
-  set( EXPECTED_CLAW_SVN_REVISION 259 )
-
   if( CLAW_FOUND )
     VERSION_GREATER_EQUAL( ${CLAW_VERSION} "1.7.0" CLAW_VERSION_OK )
 
@@ -92,14 +90,6 @@ macro(FIND_SVN_CLAW)
 	"version."
 	"Check http://libclaw.sourceforge.net/ to download the latest release."
 	)
-    else( NOT CLAW_VERSION_OK )
-      if( NOT CLAW_REVISION_NUMBER GREATER ${EXPECTED_CLAW_SVN_REVISION} )
-	message( FATAL_ERROR
- 	  "Please update and install your SVN version of libclaw for revision "
-          "greater than " ${EXPECTED_CLAW_SVN_REVISION}
-          " (got " ${CLAW_REVISION_NUMBER} ")"
- 	  )
-      endif( NOT CLAW_REVISION_NUMBER GREATER ${EXPECTED_CLAW_SVN_REVISION} ) 
     endif( NOT CLAW_VERSION_OK )
   else( CLAW_FOUND )
     message( FATAL_ERROR
