diff -up plee-the-bear-0.6.0/bear-engine/CMakeLists.txt.fpermissive plee-the-bear-0.6.0/bear-engine/CMakeLists.txt
--- plee-the-bear-0.6.0/bear-engine/CMakeLists.txt.fpermissive	2011-08-22 16:24:34.000000000 -0400
+++ plee-the-bear-0.6.0/bear-engine/CMakeLists.txt	2011-08-25 16:20:02.039211302 -0400
@@ -22,7 +22,7 @@ set( EXECUTABLE_OUTPUT_PATH "${CMAKE_CUR
 set( LIBRARY_OUTPUT_PATH "${EXECUTABLE_OUTPUT_PATH}" )
 
 # common flags
-set( CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}" )
+set( CMAKE_CXX_FLAGS "-fpermissive ${CMAKE_CXX_FLAGS}" )
 
 #-------------------------------------------------------------------------------
 # various libraries
diff -up plee-the-bear-0.6.0/bear-factory/CMakeLists.txt.fpermissive plee-the-bear-0.6.0/bear-factory/CMakeLists.txt
diff -up plee-the-bear-0.6.0/plee-the-bear/CMakeLists.txt.fpermissive plee-the-bear-0.6.0/plee-the-bear/CMakeLists.txt
--- plee-the-bear-0.6.0/plee-the-bear/CMakeLists.txt.fpermissive	2011-06-23 09:00:08.000000000 -0400
+++ plee-the-bear-0.6.0/plee-the-bear/CMakeLists.txt	2011-08-25 16:20:02.045211224 -0400
@@ -33,7 +33,7 @@ set( EXECUTABLE_OUTPUT_PATH "${CMAKE_CUR
 set( LIBRARY_OUTPUT_PATH "${EXECUTABLE_OUTPUT_PATH}" )
 
 # common flags
-set( CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}" )
+set( CMAKE_CXX_FLAGS "-fpermissive ${CMAKE_CXX_FLAGS}" )
 
 #-------------------------------------------------------------------------------
 # check SDL
