Index: kbd-1.15.3/configure.ac
===================================================================
--- kbd-1.15.3.orig/configure.ac
+++ kbd-1.15.3/configure.ac
@@ -59,7 +59,7 @@ esac
 AM_CONDITIONAL(KEYCODES_PROGS, test "$KEYCODES_PROGS" = "yes")
 
 case $host_cpu in
-	i?86*) RESIZECONS_PROGS=yes ;;
+	i?86*|x86_64) RESIZECONS_PROGS=yes ;;
 	*)     RESIZECONS_PROGS=no ;;
 esac
 AM_CONDITIONAL(RESIZECONS_PROGS, test "$RESIZECONS_PROGS" = "yes")
