diff -rupN src/plugins/azoth/plugins/acetamide/CMakeLists.txt src/plugins/azoth/plugins/acetamide/CMakeLists.txt
--- src/plugins/azoth/plugins/acetamide/CMakeLists.txt	2013-04-06 10:08:57.760772435 +0300
+++ src/plugins/azoth/plugins/acetamide/CMakeLists.txt	2013-04-06 10:08:06.000000000 +0300
@@ -20,8 +20,6 @@ SET (ACETAMIDE_SRCS
 	channelhandler.cpp
 	channelparticipantentry.cpp
 	channelpublicmessage.cpp
-	channelslistdialog.cpp
-	channelslistfilterproxymodel.cpp
 	channelsmanager.cpp
 	clientconnection.cpp
 	core.cpp
@@ -62,8 +60,6 @@ SET (ACETAMIDE_HEADERS
 	channelhandler.h
 	channelparticipantentry.h
 	channelpublicmessage.h
-	channelslistdialog.h
-	channelslistfilterproxymodel.h
 	channelsmanager.h
 	clientconnection.h
 	core.h
@@ -99,7 +95,6 @@ SET (ACETAMIDE_HEADERS
 SET (ACETAMIDE_FORMS
 	bookmarkeditwidget.ui
 	channelconfigwidget.ui
-	channelslistdialog.ui
 	ircaccountconfigurationdialog.ui
 	ircaccountconfigurationwidget.ui
 	ircjoingroupchat.ui
diff -rupN src/plugins/azoth/plugins/acetamide/ircparser.cpp src/plugins/azoth/plugins/acetamide/ircparser.cpp
--- src/plugins/azoth/plugins/acetamide/ircparser.cpp	2013-04-16 20:49:15.213000075 +0300
+++ src/plugins/azoth/plugins/acetamide/ircparser.cpp	2013-04-06 10:08:06.000000000 +0300
@@ -406,12 +395,6 @@ namespace Acetamide
 		ISH_->SendCommand (modeCmd);
 	}
 
-	void IrcParser::ChannelsListCommand (const QStringList&)
-	{
-		QString chListCmd ("LIST \r\n");
-		ISH_->SendCommand (chListCmd);
-	}
-
 	bool IrcParser::ParseMessage (const QByteArray& message)
 	{
 		QString msg;
diff -rupN src/plugins/azoth/plugins/acetamide/ircparser.h src/plugins/azoth/plugins/acetamide/ircparser.h
--- src/plugins/azoth/plugins/acetamide/ircparser.h	2013-04-16 20:49:15.219000076 +0300
+++ src/plugins/azoth/plugins/acetamide/ircparser.h	2013-04-06 10:08:06.000000000 +0300
@@ -99,7 +88,6 @@ namespace Acetamide
 		void IsonCommand (const QStringList&);
 		void QuitCommand (const QStringList&);
 		void ChanModeCommand (const QStringList&);
-		void ChannelsListCommand (const QStringList&);
 
 		bool ParseMessage (const QByteArray&);
 		IrcMessageOptions GetIrcMessageOptions () const;
diff -rupN src/plugins/azoth/plugins/acetamide/ircserverclentry.cpp src/plugins/azoth/plugins/acetamide/ircserverclentry.cpp
--- src/plugins/azoth/plugins/acetamide/ircserverclentry.cpp	2013-04-16 20:49:15.223000077 +0300
+++ src/plugins/azoth/plugins/acetamide/ircserverclentry.cpp	2013-04-06 10:08:06.000000000 +0300
@@ -48,12 +37,6 @@ namespace Acetamide
 	, ISH_ (handler)
 	, Account_ (account)
 	{
-		QAction *showChannels = new QAction (tr ("Channels list"), this);
-		connect (showChannels,
-				SIGNAL (triggered ()),
-				ISH_,
-				SLOT (showChannels ()));
-		Actions_ << showChannels;
 	}
 
 	IrcServerHandler* IrcServerCLEntry::GetIrcServerHandler () const
diff -rupN src/plugins/azoth/plugins/acetamide/ircserverhandler.cpp src/plugins/azoth/plugins/acetamide/ircserverhandler.cpp
--- src/plugins/azoth/plugins/acetamide/ircserverhandler.cpp	2013-04-29 19:39:42.729000016 +0300
+++ src/plugins/azoth/plugins/acetamide/ircserverhandler.cpp	2013-04-06 10:08:06.000000000 +0300
@@ -51,7 +40,6 @@
 #include "serverresponsemanager.h"
 #include "rplisupportparser.h"
 #include "channelsmanager.h"
-#include "channelslistdialog.h"
 
 namespace LeechCraft
 {
@@ -1181,25 +1169,6 @@ namespace Acetamide
 		ServerOptions_.IrcServerVersion_ = version;
 	}
 
-	void IrcServerHandler::GotChannelsListBegin (const IrcMessageOptions&)
-	{
-		emit gotChannelsBegin ();
-	}
-
-	void IrcServerHandler::GotChannelsList (const IrcMessageOptions& opts)
-	{
-		ChannelsDiscoverInfo info;
-		info.Topic_ = opts.Message_;
-		info.ChannelName_ = QString::fromUtf8 (opts.Parameters_.value (1).c_str ());
-		info.UsersCount_ = QString::fromUtf8 (opts.Parameters_.value (2).c_str ()).toInt ();
-		emit gotChannels (info);
-	}
-
-	void IrcServerHandler::GotChannelsListEnd (const IrcMessageOptions&)
-	{
-		emit gotChannelsEnd ();
-	}
-
 	void IrcServerHandler::connectionEstablished ()
 	{
 		ServerConnectionState_ = Connected;
@@ -1260,30 +1229,6 @@ namespace Acetamide
 		emit gotSocketError (error, socket->errorString ());
 	}
 
-	void IrcServerHandler::showChannels (const QStringList&)
-	{
-		IrcParser_->ChannelsListCommand (QStringList ());
-
-		ChannelsListDialog *dlg = new ChannelsListDialog (this);
-		dlg->setAttribute (Qt::WA_DeleteOnClose);
-		connect (this,
-				SIGNAL (gotChannelsBegin ()),
-				dlg,
-				SLOT (handleGotChannelsBegin ()),
-				Qt::UniqueConnection);
-		connect (this,
-				SIGNAL (gotChannels (ChannelsDiscoverInfo)),
-				dlg,
-				SLOT (handleGotChannels (ChannelsDiscoverInfo)),
-				Qt::UniqueConnection);
-		connect (this,
-				SIGNAL (gotChannelsEnd ()),
-				dlg,
-				SLOT (handleGotChannelsEnd ()),
-				Qt::UniqueConnection);
-		dlg->show ();
-	}
-
 	void IrcServerHandler::handleSetAutoWho ()
 	{
 		if (!XmlSettingsManager::Instance ().property ("AutoWhoRequest").toBool () &&
diff -rupN src/plugins/azoth/plugins/acetamide/ircserverhandler.h src/plugins/azoth/plugins/acetamide/ircserverhandler.h
--- src/plugins/azoth/plugins/acetamide/ircserverhandler.h	2013-04-16 20:49:15.227000075 +0300
+++ src/plugins/azoth/plugins/acetamide/ircserverhandler.h	2013-05-08 22:47:40.328000069 +0300
@@ -30,6 +30,7 @@
 #ifndef PLUGINS_AZOTH_PLUGINS_ACETAMIDE_IRCSERVERHANDLER_H
 #define PLUGINS_AZOTH_PLUGINS_ACETAMIDE_IRCSERVERHANDLER_H
 
+#include <boost/function.hpp>
 #include <QObject>
 #include <QTcpSocket>
 #include <interfaces/azoth/imessage.h>
@@ -233,11 +234,6 @@ namespace Acetamide
 				const QString& who, quint64 time);
 
 		void SetIrcServerInfo (IrcServer server, const QString& version);
-
-		void GotChannelsListBegin (const IrcMessageOptions& opts);
-		void GotChannelsList (const IrcMessageOptions& opts);
-		void GotChannelsListEnd (const IrcMessageOptions& opts);
-
 	private:
 		void SendToConsole (IMessage::Direction, const QString&);
 		void NickCmdError ();
@@ -245,7 +241,6 @@ namespace Acetamide
 	public slots:
 		void autoWhoRequest ();
 		void handleSocketError (QAbstractSocket::SocketError error);
-		void showChannels (const QStringList& = QStringList ());
 	private slots:
 		void connectionEstablished ();
 		void connectionClosed ();
@@ -259,10 +254,6 @@ namespace Acetamide
 		void nicknameConflict (const QString&);
 		void gotSocketError (QAbstractSocket::SocketError error,
 				const QString& erorString);
-
-		void gotChannelsBegin ();
-		void gotChannels (const ChannelsDiscoverInfo& info);
-		void gotChannelsEnd ();
 	};
 };
 };
diff -rupN src/plugins/azoth/plugins/acetamide/localtypes.h src/plugins/azoth/plugins/acetamide/localtypes.h
--- src/plugins/azoth/plugins/acetamide/localtypes.h	2013-04-16 20:49:15.232000077 +0300
+++ src/plugins/azoth/plugins/acetamide/localtypes.h	2013-04-06 10:08:06.000000000 +0300
@@ -67,13 +56,6 @@ namespace Acetamide
 		QString ChannelPassword_;
 	};
 
-	struct ChannelsDiscoverInfo
-	{
-		QString ChannelName_;
-		int UsersCount_;
-		QString Topic_;
-	};
-
 	enum ConnectionState
 	{
 		Connected,
diff -rupN src/plugins/azoth/plugins/acetamide/serverresponsemanager.cpp src/plugins/azoth/plugins/acetamide/serverresponsemanager.cpp
--- src/plugins/azoth/plugins/acetamide/serverresponsemanager.cpp	2013-04-16 20:49:15.246000077 +0300
+++ src/plugins/azoth/plugins/acetamide/serverresponsemanager.cpp	2013-04-06 10:08:06.000000000 +0300
@@ -234,12 +223,6 @@ namespace Acetamide
 				 this, _1);
 		Command2Action_ ["324"] = boost::bind (&ServerResponseManager::GotChannelModes,
 				 this, _1);
-		Command2Action_ ["321"] = [this] (const IrcMessageOptions& opts)
-				{ ISH_->GotChannelsListBegin (opts); };
-		Command2Action_ ["322"] = [this] (const IrcMessageOptions& opts)
-				{ ISH_->GotChannelsList (opts); };
-		Command2Action_ ["323"] = [this] (const IrcMessageOptions& opts)
-				{ ISH_->GotChannelsListEnd (opts); };
 
 		//not from rfc
 		Command2Action_ ["330"] = boost::bind (&ServerResponseManager::GotWhoIsAccount,
@@ -261,7 +244,6 @@ namespace Acetamide
 		Command2Action_ ["378"] = [this] (const IrcMessageOptions& opts)
 			{ ISH_->ShowAnswer ("278", opts.Message_); };
 
-
 		MatchString2Server_ ["unreal"] = IrcServer::UnrealIRCD;
 	}
 
diff -rupN src/plugins/azoth/plugins/acetamide/usercommandmanager.cpp src/plugins/azoth/plugins/acetamide/usercommandmanager.cpp
--- src/plugins/azoth/plugins/acetamide/usercommandmanager.cpp	2013-04-16 20:49:15.250000077 +0300
+++ src/plugins/azoth/plugins/acetamide/usercommandmanager.cpp	2013-04-06 10:08:06.000000000 +0300
@@ -199,8 +188,6 @@ namespace Acetamide
 				Parser_, _1);
 		Command2Action_ ["say"] = boost::bind (&IrcServerHandler::SayCommand,
 				ISH_, _1);
-		Command2Action_ ["list"] = boost::bind (&IrcServerHandler::showChannels,
-				ISH_, _1);
 	}
 }
 }
