diff --git a/src/ui/aboutdialog.cpp b/src/ui/aboutdialog.cpp
index 304d962..483fda8 100644
--- a/src/ui/aboutdialog.cpp
+++ b/src/ui/aboutdialog.cpp
@@ -11,7 +11,7 @@ AboutDialog::AboutDialog(QWidget *parent) :
 
     const QString buildInfo
             = QString(tr("<strong>Version:</strong> %1<br><strong>Built:</strong> %2"))
-            .arg(QCoreApplication::applicationVersion(), QString("%1 %2").arg(__DATE__, __TIME__));
+            .arg(QCoreApplication::applicationVersion(), QString("via openSUSE Build Service"));
 
     ui->buildInfoLabel->setText(buildInfo);
     ui->buttonBox->setFocus(Qt::OtherFocusReason);
