diff -urN rexloader-0.1a.rev207/Httploader/HttpLoader.pro rexloader-0.1a.rev207-patched/Httploader/HttpLoader.pro
--- rexloader-0.1a.rev207/Httploader/HttpLoader.pro	2012-02-19 01:08:11.000000000 +1100
+++ rexloader-0.1a.rev207-patched/Httploader/HttpLoader.pro	2012-02-23 21:56:37.000000000 +1100
@@ -4,6 +4,7 @@
 TARGET = HttpLoader
 TEMPLATE = lib
 CODECFORTR = UTF-8
+LIBS += -lz
 
 DESTDIR = ../usr/lib/rexloader/plugins
 
